src/ORM/Association/BelongsTo.php

```diff
@@ -151,7 +151,7 @@ public function saveAssociated(EntityInterface $entity, array $options = [])
     protected function _joinCondition($options)
     {
         $conditions = [];
-        $tAlias = $this->getTarget()->getAlias();
+        $tAlias = $this->_name;
```

Comment:
This lets the join conditions use the association's alias which can now be different than the target table.